app/controllers/api/movies_controller.rb

```diff
@@ -3,11 +3,24 @@ def index
     netflix = Rails.application.credentials.unogsng_key
 
     begin
-      response = RestClient.get('https://unogsng.p.rapidapi.com/search?type=movie&orderby=rating',
-                                headers = { 'x-rapidapi-key': netflix })
-      results = JSON.parse(response)['results'].select { |film| film['avgrating'] > 4 }
-      netflix_sorted = results.sort_by { |film| film['avgrating'] }.reverse
-      render json: { body: netflix_sorted[0..9] }, status: 200
+      if params.has_key?(:lat) && params.has_key?(:lon)
+        country_results = Geocoder.search([params[:lat], params[:lon]])
+        country = country_results.first.country
+
+        response = RestClient.get('https://unogsng.p.rapidapi.com/search?type=movie&orderby=rating',
+                                  headers = { 'x-rapidapi-key': netflix })
+        movie_array = JSON.parse(response.body)['results']
+
+        movie_results = movie_array.select { |movie| !movie['clist'].include?(country) }
+        render json: { body: movie_results[0..9]}, status: 200
+
+      elsif response = RestClient.get('https://unogsng.p.rapidapi.com/search?type=movie&orderby=rating',
+                                      headers = { 'x-rapidapi-key': netflix })
+        results = JSON.parse(response)['results'].select { |film| film['avgrating'] > 4 }
```

Comment:
You don't filter for rating if the condition is true, could be a poblem

Reply:
Done, added now to the code.